include/flatbuffers/flatbuffers.h

```diff
@@ -960,9 +960,11 @@ FLATBUFFERS_FINAL_CLASS
   }
 
   /// @brief Store a string in the buffer, which can contain any binary data.
-  /// @param[in] str A const reference to a std::string to store in the buffer.
+  /// @param[in] str A const reference to a std::string like type with support
+  /// of T::c_str and T::length() to store in the buffer.
   /// @return Returns the offset in the buffer where the string starts.
-  Offset<String> CreateString(const std::string &str) {
+  template<typename T>
+  Offset<String> CreateString(const T &str) {
```

Comment:
We need this otherwise we have to change the generated code too no?

If we were to provide a `CreateStringFrom` we have to change the code generator to know that there is a `opts.cpp_object_api_string_type` 

if you look at the converters from `nativeT -> fbb`  via `Create##Type()` it uses `CreateString()` i also don't think it will create ambiguities given the currect implementations. i.e: no other types passed work on member functions, they are just plain pointers. 

Happy to add the other `CreateStringFrom` i'm just not sure what the benefit is. 

i.e.: from a template level i don't see any errors.